scalameter-core/src/main/java/org/scalameter/execution/invocation/instrumentation/MethodInvocationCounter.java

```diff
@@ -0,0 +1,65 @@
+package org.scalameter.execution.invocation.instrumentation;
+
+import java.util.concurrent.atomic.AtomicLongArray;
+
+
+/**
+ * INTERNAL API
+ *
+ * Stores method invocations in a AtomicLongArray.
+ */
+public class MethodInvocationCounter {
+    private static volatile boolean _initialized = false;
+    private static volatile boolean _started = false;
+    private static AtomicLongArray _counts;
+
+    /**
+     * Initializes count array.
+     *
+     * @param numMethods length of an array
+     */
+    public static synchronized void setup(int numMethods) {
+        _counts = new AtomicLongArray(numMethods);
+        _initialized = true;
+    }
+
+    /**
+     * Starts measuring of method invocations.
+     */
+    public static synchronized void start() {
+        if (_initialized && !_started) {
+            _started = true;
+        }
```

Comment:
Since the Scala wrapper would also need to be accessible, I'd say that it is fine to leave this as-is.

My comment was more in the sense - why are these methods necessary? Did you want to guard against accidental invocations of `methodCalled` before/after the measurement starts/ends? If so, a comment about this intent would be great.